src/particle/particle/particle.py

```diff
@@ -1015,6 +1015,76 @@ def from_evtgen_name(cls: type[Self], name: str) -> Self:
         """
         return cls.from_pdgid(EvtGenName2PDGIDBiMap[name])
 
+    @classmethod
+    def from_nucleus_info(
+        cls: type[Self],
+        z: int,
+        a: int,
+        anti: bool = False,
+        i: int = 0,
+        l_strange: int = 0,
+    ) -> Self:
+        """
+        Get a nucleus particle from the neutron and proton count.
+        As described here https://pdg.lbl.gov/2007/reviews/montecarlorpp.pdf:
+        "To avoid ambiguities,nuclear codes should not be applied to a single hadron, like p, n
+        or Λ0, where quark-contents-based codes already exist."
+        Number of neutrons is equal to a-z.
+        PDGid format is ±10LZZZAAAI.
+
+        Parameters
+        ----------
+        z: int
+            Atomic number Z (number of protons).
+            Maximum three decimal digits.
+        a: int
+            Atomic mass number A (total number of baryons).
+            Maximum three decimal digits.
+        anti: bool, optional, defaults to False (not antimatter).
+            Whether the nucleus should be antimatter.
+        i: int, optional, default is 0
+            Isomer level I. I=0 corresponds to the ground-state.
+            I>0 are excitations.
+            Maximum is one decimal digit.
+        l_strange: int, optional, default is 0
+            Total number of strange quarks.
+            Maximum is one decimal digit.
+
+        Raises
+        ------
+        ParticleNotFound
+            If `from_pdgid`, internally called, returns no match.
+        InvalidParticle
+            If the input combination is invalid.
+        """
+        if l_strange < 0 or l_strange > 9:
+            raise InvalidParticle(
+                f"Number of strange quark l={l_strange} is invalid. Must be 0 <= l <= 9."
```

Comment:
```suggestion
                f"Number of strange quarks l={l_strange} is invalid. Must be 0 <= l <= 9."
```

Reply:
In fact all these checks with if statements, are they really necessary if you rather use the function `is_nucleus` from https://github.com/scikit-hep/particle/blob/master/src/particle/pdgid/functions.py#L232?